src/reducers/timerReducer.tsx

```diff
@@ -0,0 +1,74 @@
+import { Dispatch } from "redux";
+import { AppState } from ".";
+
+enum TypeKeys {
+    TOGGLE_TIMER = 'TOGGLE_TIMER',
+    INCREMENT_TIMER = 'INCREMENT_TIMER',
+}
+
+
+interface ToggleTimerAction {
+    type: TypeKeys.TOGGLE_TIMER;
+}
+
+interface IncrementTimerAction {
+    type: TypeKeys.INCREMENT_TIMER;
+}
+
+type ActionTypes = ToggleTimerAction | IncrementTimerAction;
+
+// setSomethingActionBuilder??
+// What is the difference between an action, and an actionbuilder?
+// normal actions, instead of an own file?
+// This now recembles more the duck.ts files
```

Comment:
ActionBuilders are pure functions that takes a parameter and returns an action of the expected type. (It's the same pattern you've used earlier. An action is just a plain object, while an action builder is the function making action creation dynamic)